src/reference/config/bindings/grpc-kafka/.partials/routes.md

```diff
@@ -0,0 +1,224 @@
+### routes\*
+
+> `array` of `object`
+
+Conditional `grpc-kafka`-specific routes for adapting `grpc` request-response streams to `kafka` topic streams.
+
+```yaml
+routes:
+  - guarded:
+      my_guard:
+        - read:messages
+    when:
+      - service: example.FanoutService
+        metadata:
+          custom-text: custom value
+          custom-binary:
+            base64: Y3VzdG9tIHZhbHVl
+    exit: kafka_cache_client
+    with:
+      capability: fetch
+      topic: messages
+      filters:
+        key: custom-key
+        headers:
+          custom-text: custom-value
+  - guarded:
+      my_guard:
+        - echo:messages
+    when:
+      - method: example.EchoService/*
+        metadata:
+          custom-text: custom value
+          custom-binary:
+            base64: Y3VzdG9tIHZhbHVl
+    exit: kafka_cache_client
+    with:
+      capability: produce
+      topic: requests
+      acks: leader_only
+      key: custom-key
+      overrides:
+        custom-text: custom-value
+      reply-to: responses
+```
+
+#### routes[].guarded
+
+> `object` as map of named: `array` of `string`
+
+Roles required by named guard.
+
+```yaml
+routes:
+  - guarded:
+      my_guard:
+        - read:messages
+```
+
+#### routes[].when
+
+> `array` of `object`
+
+List of conditions (any match) to match this route when adapting `grpc` request-response streams to `kafka` topic streams.
+Read more: [When a route matches](../../../../../concepts/bindings.md#when-a-route-matches)
+
+```yaml
+routes:
+  - when:
+      - method: example.EchoService/*
+        metadata:
+          custom-text: custom value
+          custom-binary:
+            base64: Y3VzdG9tIHZhbHVl
+```
+
+#### when[].method
+
+> `string` | Pattern: `^(?<Service>[^/]+)/(?<Method>[^/]+)`
+
+Pattern matching the fully qualified name of a `grpc` service method, in the format `<service>/<method>` allowing wildcard `*` for the method to indicate any method.
+
+#### when[].metadata
+
+> `object` as map of named: `string` or `object` properties
+
+Metadata header name value pairs (all match).
+
+Each metadata header value can be `string` or `object` with `base64` property.
+
+#### metadata.base64
+
+> `string`
+
+Base64 encoded value for binary metadata header.
+
+#### routes[].exit
+
+> `string`
+
+Next binding when following this route.
+
+```yaml
+routes:
+  - when:
+    ...
+    exit: kafka_cache_client
+```
+
+#### routes[].with\*
+
+> `object`
+
+Defines the route with the [fetch](#with-capability-fetch) capability.
+
+```yaml
+with:
+  capability: fetch
+```
+
+Defines the route with the [produce](#with-capability-produce) capability.
+
+```yaml
+with:
+  capability: produce
+```
+
+#### with.capability: fetch\*
+
+> `const`
```

Comment:
Does it make sense to include the const value here, or else just string, instead of const alone?

Reply:
No worries, I'll defer to you on that - just wanted to flag it to draw to your attention.